src/lib/dnssd/ActiveResolveAttempts.cpp

```diff
@@ -92,6 +92,22 @@ CHIP_ERROR ActiveResolveAttempts::CompleteAllBrowses()
     return CHIP_NO_ERROR;
 }
 
+void ActiveResolveAttempts::NodeIdResolutionNoLongerNeeded(const PeerId & peerId)
+{
+    for (auto & item : mRetryQueue)
+    {
+        if (item.attempt.Matches(peerId))
+        {
+            item.attempt.ConsumerRemoved();
+            if (item.attempt.ResolveData().consumerCount == 0)
```

Comment:
should this piece of code be inside ConsumeRemoved? since all of the calls (check in ResolveData and Clear) are done inside `item.attempt` it seems that we can put everything inside `ActiveResolveAttempt`

Reply:
Ah, indeed.  That is a good idea.